src/Workspaces/Core/Portable/SolutionSize/ISolutionSizeTracker.cs

```diff
@@ -1,7 +1,19 @@
-﻿namespace Microsoft.CodeAnalysis.SolutionSize
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.SolutionSize
 {
-    internal interface ISolutionSizeTracker
+    internal interface ISolutionSizeTracker : IWorkspaceService
     {
         long GetSolutionSize(Workspace workspace, SolutionId solutionId);
     }
+
+    [ExportWorkspaceService(typeof(ISolutionSizeTracker)), Shared]
+    internal class DefaultSolutionSizeTracker : ISolutionSizeTracker
+    {
+        public long GetSolutionSize(Workspace workspace, SolutionId solutionId) => -1;
+    }
```

Comment:
by default we don't compute SolutionSizes for any workspace.  This is only overridden in the VS level.